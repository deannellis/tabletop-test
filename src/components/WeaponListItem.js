import React from 'react';
import Checkbox from './Checkbox';

export default class EquipForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
        }
        this.goldPrice = this.props.price / 100;
    }
    
    render() {
        let canPurchase = (this.props.currentGold >= this.goldPrice);
        return (
            <div>   
                <h2>{this.name}</h2>
                <p>damage: {this.props.damage}</p>
                <p>price: {this.goldPrice}gp</p>
                <p>weight: {this.props.weight}</p>
                <p>{this.props.range.length && 'ranged'}</p>
                <Checkbox
                    label={`Select ${this.props.name}`}
                    handleCheckboxChange={this.toggleCheckbox}
                    key={this.props.name}
                    isDisabled={!canPurchase}
                />
            </div>    
        );
    }
    
    toggleCheckbox = isChecked => {
        this.props.handleToggleCheckbox(isChecked, this.props.id, this.goldPrice)
    }
}