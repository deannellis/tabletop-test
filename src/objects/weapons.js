// export default {
//     0000: {
//         damage: 6,
//         name: 'Hand Axe',
//         price: 400, //copper pieces
//         range: 1,
//         size: 1, // 1 = Small, 2 = Medium, 3 = Large
//         weight: 5 //pounds
//     },
//     0001: {
//         damage: 8,
//         name: 'Battle Axe',
//         price: 700, 
//         range: 1,
//         size: 2,
//         weight: 7
//     },
//     0002: {
//         damage: 10,
//         name: 'Great Axe',
//         price: 1400, 
//         range: 1,
//         size: 3,
//         weight: 10
//     },
//     0003: {
//         damage: 10,
//         name: 'Shortbow',
//         price: 2500, 
//         range: [70, 140, 210],
//         size: 2,
//         weight: 2
//     }
// }
    
export default [
    {
        id: '0000',
        damage: 6,
        name: 'Hand Axe',
        price: 400, //copper pieces
        range: 1,
        size: 1, // 1 = Small, 2 = Medium, 3 = Large
        weight: 5 //pounds
    },
    {
        id: '0001',
        damage: 8,
        name: 'Battle Axe',
        price: 700, 
        range: 1,
        size: 2,
        weight: 7
    },
    {
        id: '0002',
        damage: 10,
        name: 'Great Axe',
        price: 1400, 
        range: 1,
        size: 3,
        weight: 10
    },
    {
        id: '0003',
        damage: 10,
        name: 'Shortbow',
        price: 2500, 
        range: [70, 140, 210],
        size: 2,
        weight: 2
    }
]

// const sampleWeapon = {
//     id: 0000,
//     price: 1000, //copper pieces
//     weight: 5, //pounds
//     name: 'hand axe',
//     size: 1, // 1 = Small, 2 = Medium, 3 = Large
//     damage: 6
// }