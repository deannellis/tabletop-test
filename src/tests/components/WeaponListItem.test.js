/* global expect */
import React from 'react';
import { shallow } from 'enzyme';
import WeaponListItem from '../../components/WeaponListItem';
import weapons from '../../objects/weapons';

test('should render weapon that can be purchased', () => {
    const wrapper = shallow(<WeaponListItem { ...weapons[0] } currentGold={450} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render weapon that cannot be purchased (checkbox disabled)', () => {
    const wrapper = shallow(<WeaponListItem { ...weapons[0] } currentGold={0} />);
    expect(wrapper).toMatchSnapshot();
});