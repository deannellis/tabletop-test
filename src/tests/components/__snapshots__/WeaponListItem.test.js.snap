// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render weapon that can be purchased 1`] = `
<div>
  <h2>
    Hand Axe
  </h2>
  <p>
    damage: 
    6
  </p>
  <p>
    price: 
    4
    gp
  </p>
  <p>
    weight: 
    5
  </p>
  <p />
  <Checkbox
    handleCheckboxChange={[Function]}
    isDisabled={false}
    key="Hand Axe"
    label="Select Hand Axe"
  />
</div>
`;

exports[`should render weapon that cannot be purchased (checkbox disabled) 1`] = `
<div>
  <h2>
    Hand Axe
  </h2>
  <p>
    damage: 
    6
  </p>
  <p>
    price: 
    4
    gp
  </p>
  <p>
    weight: 
    5
  </p>
  <p />
  <Checkbox
    handleCheckboxChange={[Function]}
    isDisabled={true}
    key="Hand Axe"
    label="Select Hand Axe"
  />
</div>
`;
